backup/restore: hardcode MD5 algo for format v2/v3 decryption

rustybird opened the pull request.

marmarek commented:
Please update tests too, to generate backup as it used to be with the old openssl.
(tests currently aren't started automatically, but I hope the issue will be resolved soon)

rustybird committed:
backup/restore: hardcode MD5 algo for format v2/v3 decryption

Old OpenSSL versions (which were used to encrypt format v2 and v3
backups) defaulted to MD5 for key derivation from the password.

OpenSSL 1.1.0+ in R4.1+ changed that default (not only for encryption
but also for *decryption*) to SHA256:

https://github.com/openssl/openssl/commit/f8547f62c212837dbf44fb7e2755e5774a59a57b#diff-5a32e43870f189e8abf5e1b388115b001c9dadc23e582f28b667738857c237b6

Fixes QubesOS/qubes-issues#7852

rustybird commented:
Done. Missed it because I didn't set `ENABLE_SLOW_TESTS=1`
